src/base.ts

```diff
@@ -3,5 +3,6 @@ import {Command} from '@oclif/core';
 export default abstract class extends Command {
   async catch(e: any) {
     console.error(`${e.name}: ${e.message}`);
+    process.exitCode = 1;
```

Comment:
Yeah, this solves it 👍🏼 , but wait for @derberg's review.

Reply:
>  and if we throw the error oclif has a global error handler which would handle it, but it is for stuff like command not found

if this is only for missing commands, then I don't think it will do the trick. Related issue talks about exit code from `asyncapi validate` when validation fails. Shouldn't it be done in `validate.ts` then?

Reply:
I think that https://oclif.io/docs/commands#thiserrormessage-string--error-options-code-string-exit-number-ref-string-suggestions-string will be better :)

Reply:
@Souvikns Yeah, that solutions works fine 👌  I also tried to apply what @magicmatatjahu proposed and it works as well. The output is not exactly the same though. I mean, when the line below is added to the `catch`:

```
this.error(`${e.name}: ${e.message}`, {exit: 1});
```
Then we get the output as follows:
```
 ›   Error: ValidationError: Document can't be null or falsey.
1
```
Whereas by applying the first solution proposed, the output is:
```
ValidationError: Document can't be null or falsey.
1
```

I guess we should opt for the solution already included in this PR, since the output displayed by the alternative solution seems to be redundant . What do you think, guys? 🤔

Reply:
@peter-rr @magicmatatjahu 

I would actually prefer the solution that is described in the ocliff documentation (the first output), the `Error...` display is not a problem for me.

Reply:
I also think we should use the solution described in oclif, and we can even change the error message and not have the `Error...` display. 

I think following https://oclif.io/docs/error_handling#binrun-catch-handler we can just write our own custom error handler function in the `bin/run` file.

Reply:
I opt for that :) @peter-rr Could you propagate suggestion? :)

Reply:
Sure! Let me work on it 💪

Reply:
After applying the solution as described in oclif and also adding the error handling to the `bin/run` file, the output remains the same with the `Error...` display.

I needed to install the `@oclif/error` module since it wasn't included yet in package.json, and finally found out this libray has been replaced by `@oclif/core` 👉  https://github.com/oclif/errors. So I guess there must be any conflict with those libraries and I'm missing something when editing the `bin/run` file 🤔 Any clue about it? @Souvikns @magicmatatjahu

Reply:
I don't think you need to install `oclif/error` anymore. From newer versions everything is under `oclif/core`. 😄 I think they did not update their documentation, If you see the latest changes in our branch you will see 

```js
#!/usr/bin/env node

const oclif = require('@oclif/core')

oclif.run().then(require('@oclif/core/flush')).catch(require('@oclif/core/handle'))
```

see there is `@oclif/core/handle`, you don't have to install anything else. 

You can just write a custom handler function like 

```js
#!/usr/bin/env node

const oclif = require('@oclif/core')

oclif.run().then(require('@oclif/core/flush')).catch(err => {
  console.error(err.message)
})
```

Reply:
After applying the latest changes proposed, I'm getting the following output:
```
ValidationError: Document can't be null or falsey.

1
```
However, 10 tests from several commands are failing due to the `Error: ` title is now added to every error and that's something not expected by the tests. Should I update those tests to make them pass? Or is it possible to remove or ignore somehow that `Error: ` title to avoid the tests fail? 🤔

Reply:
I tested and I think this is coming from `dev.cmd` file in `bin` folder which is where I guess the tests are running. If you do `./bin/run validate wrong/path` then you get `error loading AsyncAPI document from file: ./something.yaml is an invalid file path` but when we do `./bin/dev validate wrong/path` we get 

```
Error: error loading AsyncAPI document from file: ./something.yaml is an invalid file path
    at Object.error (/Users/postman/Documents/GitHub/cli/node_modules/@oclif/core/lib/errors/index.js:29:15)
    at Validate.error (/Users/postman/Documents/GitHub/cli/node_modules/@oclif/core/lib/command.js:90:23)
    at Validate.catch (/Users/postman/Documents/GitHub/cli/src/base.ts:5:10)
    at Validate._run (/Users/postman/Documents/GitHub/cli/node_modules/@oclif/core/lib/command.js:71:29)
    at async Config.runCommand (/Users/postman/Documents/GitHub/cli/node_modules/@oclif/core/lib/config/config.js:226:25)
    at async Object.run (/Users/postman/Documents/GitHub/cli/node_modules/@oclif/core/lib/main.js:73:5)
```